### PR TITLE
various id mapper fixes

### PR DESCRIPTION
**Pull request by JoshuaVSherman**

### CODE REVIEW

**Comment by JoshuaVSherman** [comment]
https://github.com/PATRIC3/patric3_website/issues/1883

https://github.com/PATRIC3/patric3_website/issues/1864

File: public/js/p3/widget/templates/IDMapping.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	<tbody>
 		<tr><th class="idMappingHeader">PATRIC Identifiers</th><th class="idMappingHeader" >REFSEQ Identifiers</th></tr>
 		<tr><td rel="patric_id">PATRIC ID</td><td rel="refseq_locus_tag">RefSeq Locus Tag</td></tr>
-		<tr><td rel="feature_id" >Feature ID</td><td rel="protein_id">RefSeq</td></tr>
+		<tr><td rel="feature_id" >Feature ID</td><td rel="protein_id">Protein ID</td></tr>
```
Comment (hyoo) [inline public/js/p3/widget/templates/IDMapping.html]: change it to "RefSeq (Protein ID)"

Comment (JoshuaVSherman) [reply public/js/p3/widget/templates/IDMapping.html]: @hyoo done

File: public/js/p3/store/IDMappingAppMemoryStore.js [hunk 2]
```diff
@@ -141,9 +141,9 @@ define([
         else if (via =="refseq_locus_tag"){
           joinId = {"genome_feature":"refseq_locus_tag","id_ref":"Gene_OrderedLocusName"};
         }
-        else if (via =="protein_id"){
-          joinId = {"genome_feature":"protein_id","id_ref":"RefSeq"};
-        }
+        // else if (via =="protein_id"){
+        //   joinId = {"genome_feature":"protein_id","id_ref":"RefSeq"};
+        // }
```
Comment (hyoo) [inline public/js/p3/store/IDMappingAppMemoryStore.js]: This block, i believe, is handling mapping strategy (advanced option in the form). So, we should keep it, i guess.

File: public/js/p3/store/IDMappingAppMemoryStore.js [hunk 3]
```diff
@@ -161,7 +161,7 @@ define([
         // console.log(this.state);
 
         if(fromIdGroup === 'PATRIC'){
-          if(toIdGroup === 'PATRIC'){
+          if(toIdGroup === 'PATRIC' || toId === 'protein_id'){
```
Comment (hyoo) [inline public/js/p3/store/IDMappingAppMemoryStore.js]: as far as i remember, there is a map to control what should be considered which "toIdGroup". Adding or in here will make it difficult to trace later.

File: public/js/p3/widget/app/IDMapper.js [hunk 4]
```diff
@@ -82,6 +82,10 @@ define([
     */
     return true;
   },
+  reset: function(){
+    console.log('going to clear the form');
```
Comment (hyoo) [inline public/js/p3/widget/app/IDMapper.js]: please remove tracing logs, once you know how it works.

File: public/js/p3/store/IDMappingAppMemoryStore.js (hunk 3 above)
Comment (hyoo) [reply public/js/p3/store/IDMappingAppMemoryStore.js]: https://github.com/PATRIC3/p3_web/blob/develop/public/js/p3/widget/app/IDMapper.js#L129

**Comment by JoshuaVSherman** [comment]
@hyoo done

**Pull request merged**